numba/tests/test_linalg.py

```diff
@@ -361,6 +382,49 @@ def assert_no_nan_or_inf(self, cfunc, args):
         msg = "Array must not contain infs or NaNs."
         self.assert_error(cfunc, args, msg, np.linalg.LinAlgError)
 
+    def assert_contig_sanity(self, got, expected_contig):
+        """
+        This checks that in a computed result from numba (array, possibly tuple
+        of arrays) all the arrays are contiguous in memory and that they are
+        all at least one of "C_CONTIGUOUS" or "F_CONTIGUOUS". The computed
+        result of the contiguousness is then compared against a hardcoded
+        expected result.
+
+        got: is the computed results from numba
+        expected_contig: is "C" or "F" and is the expected type of
+                        contiguousness across all input values
+                        (and therefore tests).
+        """
+        flgc = "C_CONTIGUOUS"
+        flgf = "F_CONTIGUOUS"
+
+        if isinstance(got, tuple):
+            # tuple present, check all results
+            # find the contiguousness of the first result
+            c_contig = got[0].flags[flgc]
+            f_contig = got[0].flags[flgf]
+            # AND in the flags from other possible results
+            for k in range(1, len(got)):
```

Comment:
... otherwise you can also use a set, e.g. `{a.flags.c_contiguous for a in got}` and then compare that set to either `{True}` or `{False}`.